File: grassseed.rb
cost = gets.to_f
if cost > 0 and cost <=100
   n = gets.to_i
   if n > 0 and n <= 100
       ans = 0
       for i in (0...n)
          l,w = gets.chomp.split.map(&:to_f)
          ans = ans + (l * w)
       end
       print("%.7f"%[ans*cost])
    end
end

File: grassseed.cpp
#include<iostream>
#include<stdio.h>
using namespace std;
int main(){
  double cost,ans = 0;
  int n;
  cin >> cost;
  if(cost > 0 && cost <= 100){
    cin >> n;
    if(n > 0 && n <= 100){
      double l,w;
      for(int i = 0;i < n;i++){
      cin >> l >> w;
      ans = ans + (l*w);
      }
      printf("%.7f",ans*cost);
    }

  }
}